src/lib.rs

```diff
@@ -443,6 +462,15 @@ impl Game {
                     }
                 }
             }
+            Message::FinishedTask(FinishedTask { color, index }) => {
```

Comment:
I don't quite get what is going on here. It looks like a call (`Message::FinishedTask(...)`) but then has a fat arrow and a block with some stuff inside. As a guess, I might think this performs some network side-effect and then calls back and does whatever work is in the block. But, that feels like a wild guess. I could probably grok it better if I read wherever `FinishedTask` is defined..

Reply:
Okay, I read that `Message::FinishedTask` is an enum, so actually this looks more like a call signature for an anonymous function. And, unfurling the code above, I now see we are in a `match` clause. So, I get it 😅 I think...

Reply:
Yeah, you got it, this is a pattern to match that also pulls `color` and `index` into local variables in the case where the incoming message is a FinishedTask message